src/lsqecc/pauli_rotations/rotation.py

```diff
@@ -192,6 +192,43 @@ def from_list(pauli_ops: List[PauliOperator], rotation: Fraction) -> "PauliRotat
             r.change_single_op(i, op)
         return r
 
+    def remove_y_operators(self) -> Tuple[List["PauliRotation"], List["PauliRotation"]]:
+        """Remove Y operators from a Pauli Rotation. Currently supports only pi/8 rotation"""
+        if self.rotation_amount not in (Fraction(1, 8), Fraction(-1, 8)):
+            raise NotImplementedError("Method only supports pi/8 rotations")
+
+        left_rotations: List["PauliRotation"] = list()
+        right_rotations: List["PauliRotation"] = list()
+        y_op_indices = list()
+
+        # Find Y operators and modify them into X operators
+        for index, operator in enumerate(self.ops_list):
+            if operator == PauliOperator.Y:
+                y_op_indices.append(index)
+                self.ops_list[index] = PauliOperator.X
+
+        if not y_op_indices:
+            return left_rotations, right_rotations
+
+        if len(y_op_indices) % 2 == 0:
+            # For even numbers of Y operators, add 2 additional pi/4 rotations (one on each side)
+            first_y_operator = y_op_indices.pop(0)
+            new_rotation_ops = [
+                PauliOperator.Z if i == first_y_operator else PauliOperator.I
+                for i in range(self.qubit_num)
+            ]
+            left_rotations.append(PauliRotation.from_list(new_rotation_ops, Fraction(1, 4)))
+            right_rotations.append(PauliRotation.from_list(new_rotation_ops, Fraction(-1, 4)))
+
+        new_rotation_ops = [
+            PauliOperator.Z if i in y_op_indices else PauliOperator.I for i in range(self.qubit_num)
+        ]
+
+        left_rotations.append(PauliRotation.from_list(new_rotation_ops, Fraction(1, 4)))
+        right_rotations.append(PauliRotation.from_list(new_rotation_ops, Fraction(-1, 4)))
+
+        return left_rotations, right_rotations
```

Comment:
What is the workflow here? Adding the left and right rotations into the circuit will take place in `circuit.py` ?

Reply:
Yep, basically:
```
# Before
---rotation---

# After
---left_rotations---rotation_without_Y---right_rotations---
```

I will make it clearer in the code and docstring later.

Reply:
There is some discussion in this issue https://github.com/latticesurgery-com/lattice-surgery-compiler/issues/35:

From Litinski's GoSC:
![image](https://user-images.githubusercontent.com/36427091/110456550-8d500b80-807e-11eb-9989-51276e316ec5.png)

Reply:
With my suggestion above the workflow would become to replace the rotations with the Y operator with the list returned by `rotation.get_y_operator_free_equivalent`

Reply:
Implemented @gwwatkin's suggestion